### PR TITLE
net: do wii network transfers in blocks

### PR DESCRIPTION
**Pull request by DacoTaco**
brought to light that it should be recommended to prevent things like #177 
tested using priiloader's update feature

### CODE REVIEW

File: libogc/network_wii.c [hunk 1]
```diff
@@ -885,40 +885,56 @@ s32 net_send(s32 s, const void *data, s32 size, u32 flags)
 s32 net_sendto(s32 s, const void *data, s32 len, u32 flags, struct sockaddr *to, socklen_t tolen)
 {
 	s32 ret;
+	const s32 maxblocksize = (NET_HEAP_SIZE-64);
```
Comment (WinterMute) [inline libogc/network_wii.c:888]: I would drop this to NET_HEAP_SIZE/2 to try and avoid failure in the net_mallocs.

I'd also try this as a static const at global scope & see what the code gen is like.

File: libogc/network_wii.c [hunk 2]
```diff
@@ -885,40 +885,56 @@ s32 net_send(s32 s, const void *data, s32 size, u32 flags)
...
 	if (net_ip_top_fd < 0) return -ENXIO;
 	if (tolen > 28) return -EOVERFLOW;
+	
+	for(ret = 0; ret < len;)
+	{
+		s32 send_size = len - ret;
+		if(send_size > maxblocksize)
+			send_size = maxblocksize;
+		
+		message_buf = net_malloc(send_size);
```
Comment (WinterMute) [inline libogc/network_wii.c:901]: move this allocation outside the loop.

File: libogc/network_wii.c [hunk 3]
```diff
@@ -885,40 +885,56 @@ s32 net_send(s32 s, const void *data, s32 size, u32 flags)
...
+		debug_printf("net_send retuned %d\n", sent_size);
+		
+		if(sent_size <= 0)
+		{
+			ret = sent_size;
+			break;
+		}
+		
+		ret += sent_size;
+		if(message_buf!=NULL) net_free(message_buf);
```
Comment (WinterMute) [inline libogc/network_wii.c:935]: and the free should be outside the loop too.

File: libogc/network_wii.c [hunk 4]
```diff
@@ -931,6 +947,7 @@ s32 net_recv(s32 s, void *mem, s32 len, u32 flags)
 s32 net_recvfrom(s32 s, void *mem, s32 len, u32 flags, struct sockaddr *from, socklen_t *fromlen)
 {
 	s32 ret;
+	const s32 maxblocksize = (NET_HEAP_SIZE-64);
```
Comment (WinterMute) [inline libogc/network_wii.c:950]: This is why I'd prefer having this const at global scope so it doesn't need repeating in multiple places.

File: libogc/network_wii.c [hunk 5]
```diff
@@ -941,35 +958,57 @@ s32 net_recvfrom(s32 s, void *mem, s32 len, u32 flags, struct sockaddr *from, so
...
 		from->sa_len = *fromlen;
 	}
+	
+	for(ret = 0; ret < len;)
+	{
+		s32 recv_size = len - ret;
+		if(recv_size > maxblocksize)
+			recv_size = maxblocksize;
+		
+		message_buf = net_malloc(recv_size);
```
Comment (WinterMute) [inline libogc/network_wii.c:968]: and again, malloc outside loop.

File: libogc/network_wii.c [hunk 6]
```diff
@@ -941,35 +958,57 @@ s32 net_recvfrom(s32 s, void *mem, s32 len, u32 flags, struct sockaddr *from, so
...
 			ret = -EOVERFLOW;
-			goto done;
+			break;
 		}
-
-		memcpy(mem, message_buf, ret);
+		
+		if (fromlen && from) *fromlen = from->sa_len;	
+		memcpy(mem+ret, message_buf, received);
+		net_free(message_buf);
```
Comment (WinterMute) [inline libogc/network_wii.c:1000]: and corresponding free outside the loop.

**Review by WinterMute — CHANGES_REQUESTED** [review]

File: libogc/network_wii.c (hunk 1 above)
Comment (DacoTaco) [reply libogc/network_wii.c]: moved to top of file :)

File: libogc/network_wii.c (hunk 2 above)
Comment (DacoTaco) [reply libogc/network_wii.c]: moved outside loop with max of either len or maxblocksize

**Pull request merged**